acceptance/tests/facts/windows_os.rb

```diff
@@ -0,0 +1,65 @@
+test_name "Test facter os reports correct os.windows.system32" do
+
+#  disable this test until changes in beaker 
+#  https://tickets.puppetlabs.com/browse/BKR-1627
+#  will allow this test to run in timely manner
+  confine :to, :platform => /no-platform/
+#  confine :to, :platform => /windows/
+#  confine :except, :platform => /2008/
+
+  agents.each do |agent|
+    os_type = on(agent, powershell("facter os.windows.installation_type")).stdout.chomp
+    domain_firewall_state = on(agent, powershell("'(Get-NetFirewallProfile -Profile Domain).Enabled'")).stdout.chomp
+    public_firewall_state = on(agent, powershell("'(Get-NetFirewallProfile -Profile Public).Enabled'")).stdout.chomp
+    private_firewall_state = on(agent, powershell("'(Get-NetFirewallProfile -Profile Private).Enabled'")).stdout.chomp
+    on(agent, powershell("'Set-NetFirewallProfile -Profile Domain,Public,Private -Enabled False'"))
+
+    teardown do
+      if os_type == "Server" && on(agent, powershell("'Import-module servermanager ; (Get-WindowsFeature -name RDS-RD-Server).Installed'")).stdout.chomp == "True"
+        on(agent, powershell("'Import-module servermanager ; Remove-WindowsFeature –Name RDS-RD-Server'"))
+        agent.reboot
+        #assert_equal(true, agent.down?, 'Cound not remove RDS-RD-Server, failed to bring the host down')
+        begin
+          agent.down?
+        rescue
+          #TODO: Handle restart pending
+          puts 'Assuming that restart was not caught'
+        end
+        agent.wait_for_port(3389)
```

Comment:
as remote server is removed, should we still wait for port 3389?

Reply:
Yes, server is removed, but default remote desktop capabilities built in  in every Windows OS are still installed; waiting for this to be up because , port 22 is available too early, when Windows is to fully loaded